src/main/java/com/company/demoapplication/TaskPublisher.java

```diff
@@ -55,7 +55,7 @@ private static SendMessageRequest createRequest(String sqsQueueUrl, String key)
     }
 
     private static List<String> listImagesOnS3() {
-        List<S3ObjectSummary> images = Main.s3Client().listObjectsV2(new ListObjectsV2Request().withBucketName(Main.bucketName).withPrefix(Main.sampleImagesFolder)).getObjectSummaries();
+        List<S3ObjectSummary> images = Main.s3Client().listObjects(new ListObjectsRequest().withBucketName(Main.bucketName).withPrefix(Main.sampleImagesFolder)).getObjectSummaries();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This code might not produce accurate results if the operation returns paginated results instead of all results. Consider adding another call to check for additional results.